spring-pulsar/src/test/java/org/springframework/pulsar/listener/PulsarListenerTests.java

```diff
@@ -955,4 +957,53 @@ void pulsarMessagesBatchListenerWithHeaders(Messages<String> data,
 
 	}
 
+	@Nested
+	@ContextConfiguration(classes = ConsumerPauseTest.ConsumerPauseConfig.class)
+	class ConsumerPauseTest {
+
+		private static final CountDownLatch latch = new CountDownLatch(10);
+
+		@Autowired
+		PulsarListenerEndpointRegistry pulsarListenerEndpointRegistry;
+
+		@Test
+		void containerPauseAndResumeSuccessfully() throws Exception {
+			for (int i = 0; i < 3; i++) {
+				pulsarTemplate.send("consumer-pause-topic", "hello-" + i);
+			}
+			// wait until all 3 messages are received by the listener
+			Awaitility.await().timeout(Duration.ofSeconds(10)).until(() -> latch.getCount() == 7);
+			PulsarMessageListenerContainer container = pulsarListenerEndpointRegistry
+					.getListenerContainer("consumerPauseListener");
+			if (container != null) {
+				container.pause();
+			}
+			Thread.sleep(1000);
+			for (int i = 3; i < 10; i++) {
+				pulsarTemplate.send("consumer-pause-topic", "hello-" + i);
+			}
+			Thread.sleep(1000);
+			assertThat(latch.getCount()).isEqualTo(7);
+
+			if (container != null) {
+				container.resume();
+			}
+			// All latch must be received by now
+			assertThat(latch.await(10, TimeUnit.SECONDS)).isTrue();
+		}
+
+		@EnablePulsar
+		@Configuration
+		static class ConsumerPauseConfig {
+
+			@PulsarListener(id = "consumerPauseListener", subscriptionName = "consumer-pause-subscription",
+					topics = "consumer-pause-topic", properties = { "receiverQueueSize=1" })
```

Comment:
Do we need to mention anything about the `receiverQueueSize` in the docs? I think because we control the state in the container, we do not have to enforce this. I am guessing you are just doing that here to aid in the test?

Reply:
Good point to add this to the docs (just as a passing reference).

Reply:
But i will do that in a subsequent PR though.